sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs

```diff
@@ -554,63 +528,22 @@ private async Task<SearchResources> CreateSearchServiceIndexAndDocumentsAsync()
         /// </summary>
         /// <returns>The current <see cref="SearchResources"/>.</returns>
-        private async Task<SearchResources> CreateStorageAccountAsync()
+        private async Task<SearchResources> CreateHotelsBlobContainerAsync()
         {
             if (TestFixture.Mode != RecordedTestMode.Playback)
             {
-                using CancellationTokenSource cts = new CancellationTokenSource(Settings.Timeout);
-
-                StorageManagementClient client = GetStorageManagementClient();
-
-                // Use same resource name but without dashes that are invalid for storage account names.
-                StorageAccountName = Settings.ResourcePrefix.Replace("-", string.Empty) + Settings.Random.GetName(8);
-                StorageAccountCreateParameters parameters = new StorageAccountCreateParameters
-                {
-                    AccessTier = AccessTier.Hot,
-                    Kind = Microsoft.Azure.Management.Storage.Models.Kind.BlobStorage,
-                    Location = Settings.Location,
-                    Sku = new Microsoft.Azure.Management.Storage.Models.Sku
-                    {
-                        Name = Microsoft.Azure.Management.Storage.Models.SkuName.StandardLRS,
-                    },
-                };
-
-                StorageAccount account = await client.StorageAccounts.CreateAsync(
-                    Settings.ResourceGroup,
-                    StorageAccountName,
-                    parameters,
-                    cts.Token);
-
-                RequiresStorageCleanup = true;
-
-                StorageAccountListKeysResult keys = await client.StorageAccounts.ListKeysAsync(
-                    Settings.ResourceGroup,
-                    account.Name,
-                    cts.Token);
-                StorageAccountKey key = keys.Keys[0];
+                BlobContainerName = Settings.Random.GetName(8);
 
-                StorageAccountConnectionString = $"DefaultEndpointsProtocol=https;AccountName={account.Name};AccountKey={key.Value};EndpointSuffix=core.windows.net";
-            }
-
-            return this;
-        }
-
-        /// <summary>
-        /// Upload <see cref="TestDocuments"/> to a new blob storage container named "hotels".
-        /// </summary>
-        /// <returns>The current <see cref="SearchResources"/>.</returns>
-        private async Task<SearchResources> CreateHotelsContainerAsync()
-        {
-            if (TestFixture.Mode != RecordedTestMode.Playback)
-            {
                 using CancellationTokenSource cts = new CancellationTokenSource(Settings.Timeout);
 
-                BlobContainerClient client = new BlobContainerClient(StorageAccountConnectionString, "hotels");
+                BlobContainerClient client = new BlobContainerClient(StorageAccountConnectionString, BlobContainerName);
                 await client.CreateAsync(cancellationToken: cts.Token);
```

Comment:
Should probably make this `CreateIfNotExistsAsync` instead.  It's safer for edge cases where a network failure causes a retry even though the service received and responded.